skins/cat17/address-form/src/__tests__/store.spec.ts

```diff
@@ -0,0 +1,338 @@
+				null
+				) ).toStrictEqual( ['title', 'street', 'city'] );
+		});
+		
+	} );
+	
+	describe( 'Getters/allFieldsAreValid', () => {
+		it( 'asserts all fields are valid on initialization', () => {
+			const invalidFields: Array<String> = [];
+			expect( getters.allFieldsAreValid(
+				newMinimalStore( {} ),
+				{ invalidFields },
+				null,
+				null
+				) ).toBe( true );
+		} );
+		it( 'asserts not all fields are valid when there are invalid fields', () => {
+			const state = {
+				form: {
+					title: Validity.INVALID,
+					street: Validity.INVALID,
+					city: Validity.INVALID
+				}
+			}
+			const invalidFields: Array<String> = ['title', 'street', 'city'];
+			expect( getters.allFieldsAreValid(
+				newMinimalStore( { state } ),
+				{ invalidFields },
+				null,
+				null
+				) ).toBe( false );
+		} );
+	} );
+
+	describe( 'Actions/validateInput', () => {
+		it( 'commits to mutation [VALIDATE_INPUT] with the correct field', () => {
+			const commit = jest.fn();
+			const action = actions[ 'validateInput' ] as any;
+			action( { commit }, 'postcode' );
+			expect( commit ).toBeCalledWith(
+				'VALIDATE_INPUT',
+				'postcode'
+			)
+		} )
+	} );
+
+	describe( 'Actions/storeAddressFields', () => {
+		const fakeFormData = {
+			salutation: {
+				name: 'salutation',
+				value: '',
+				pattern: '^(Herr|Frau)$',
+				optionalField: false
+			},
+			title: {
+				name: 'title',
+				value: '',
+				pattern: '',
+				optionalField: false
+			},
+			companyName: {
+				name: 'companyName',
+				value: '',
+				pattern: '^.+$',
+				optionalField: true
+			},
+			firstName: {
+				name: 'firstName',
+				value: '',
+				pattern: '^.+$',
+				optionalField: false
+			},
+			lastName: {
+				name: 'lastName',
+				value: '',
+				pattern: '^.+$',
+				optionalField: false
+			},
+			street: {
+				name: 'street',
+				value: '',
+				pattern: '^.+$',
+				optionalField: false
+			},
+			city: {
+				name: 'city',
+				value: '',
+				pattern: '^.+$',
+				optionalField: false
+			},
+			postcode: {
+				name: 'postcode',
+				value: '',
+				pattern: '[0-9]{4,5}$',
+				optionalField: false
+			},
+			country: {
+				name: 'country',
+				value: 'DE',
+				pattern: '',
+				optionalField: false
+			},
+			addressType: {
+				name: 'addressType',
+				value: 'person',
+				pattern: '',
+				optionalField: false
+			}
+		},
+		 payload = {
+			transport: {
+				postData: jest.fn( function( validationResult ) { return Promise.resolve() } )
+			},
+			validateAddressURL: '',
+			formData: fakeFormData
+		};
+
+		it( 'commits to mutation [MARK_EMPTY_FIELD_INVALID] with form data', () => {
+			const commit = jest.fn( function() { return Promise.resolve() } );
+			const action = actions[ 'storeAddressFields' ] as any;
+			action( { commit, getters }, payload );
+			expect( commit ).toBeCalledWith(
+				'MARK_EMPTY_FIELD_INVALID',
+				fakeFormData
+			);
+		} );
+
+		it( 'does not start address validation if there are invalid fields', () => {
```

Comment:
I know that incomplete fields are considered invalid inside this check, so technically this is correct, but reading it without that knowledge made me wonder why address validation is not started when there are invalid fields. And how fields can ever become valid when the validation is never started once they become invalid.

Reply:
Good that you mentioned it because while changing the description I noticed the next test has the exact same setup but the two assert different things. So something's wrong..

Reply:
Fixed ⚒